var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -90,33 +89,73 @@ class IntelParallelStudio(IntelInstaller):
-    variant('all', default=False,
-            description="Install all files with the requested edition")
-    variant('mpi', default=True,
-            description="Install the Intel MPI library and ITAC tool")
-    variant('mkl', default=True, description="Install the Intel MKL library")
-    variant('daal',
-            default=True, description="Install the Intel DAAL libraries")
-    variant('ipp', default=True, description="Install the Intel IPP libraries")
-    variant('tools', default=True, description="Install the Intel Advisor, "
-            "VTune Amplifier, and Inspector tools")
-
-    variant('shared', default=True, description='Builds shared library')
-    variant('ilp64', default=False, description='64 bit integers')
-    variant('openmp', default=False, description='OpenMP multithreading layer')
-
-    provides('mpi', when='@cluster.0:cluster.9999+mpi')
-    provides('mkl', when='+mkl')
+            description='Allow use of --enable-new-dtags in MPI wrappers')
+    variant('shared',   default=True,
+            description='Builds shared library')
+    variant('ilp64',    default=False,
+            description='64 bit integers')
+    variant('openmp',   default=False,
+            description='OpenMP multithreading layer')
+    variant('all',      default=False,
+            description='Install all components of the requested edition')
+
+    # Components available in all editions
+    variant('daal', default=True,
+            description='Install the Intel DAAL libraries')
+    variant('gdb',  default=False,
+            description='Install the Intel Debugger for Heterogeneous Compute')
+    variant('ipp',  default=True,
+            description='Install the Intel IPP libraries')
+    variant('mkl',  default=True,
+            description='Install the Intel MKL library')
+    variant('mpi',  default=True,
+            description='Install the Intel MPI library')
+    variant('tbb',  default=True,
+            description='Install the Intel TBB libraries')
+
+    # Components only available in the Professional and Cluster Editions
+    variant('advisor',   default=False,
+            description='Install the Intel Advisor')
+    variant('clck',      default=False,
+            description='Install the Intel Cluster Checker')
+    variant('inspector', default=False,
+            description='Install the Intel Inspector')
+    variant('itac',      default=False,
+            description='Install the Intel Trace Analyzer and Collector')
+    variant('vtune',     default=False,
+            description='Install the Intel VTune Amplifier XE')
+
     provides('daal', when='+daal')
-    provides('ipp', when='+ipp')
+    provides('daal', when='+all')
 
-    # virtual dependency
-    provides('blas', when='+mkl')
-    provides('lapack', when='+mkl')
+    provides('ipp', when='+ipp')
+    provides('ipp', when='+all')
+
+    provides('mkl',       when='+mkl')
+    provides('mkl',       when='+all')
```

Comment:
i would say everyone's on board with removing `all` as it is covered by separate components.